app/lib/widgets/cards/productCard/shared/price_discount.dart

```diff
@@ -0,0 +1,225 @@
+///
+/// FlexPriceDiscount implemented with same interface as FlexPrice with additional functionality
+/// - Conditional rendering of Current Price as a 'sale' price when an oldPrice is provided
+/// - When no 'oldPrice' is present, fallback to standard ``FlexPrice``
+/// - Theme customization for Price, Discount Price & Old Price values
+/// - Price & Old Price Semantic Label Optional Parameters
+/// - single line or line Wrap parameter for Price Value overflows
+///
+/// Parameters
+/// - [price]: Current Product Price, when [oldPrice] provided, will be styled as a 'sale price'
+/// - [oldPrice]: Previous Product Price. Default Text Styling linethrough
+/// - [priceFormatter]: Price Formatter Callback provided to underlying FlexPrice Widget
+/// - [priceStyle]: Override Price TextStyle for Current Price, Either when oldPrice is provided or not.
+/// - [oldPriceStyle]: Override [oldPrice] TextStyle (Default Strikethrough) FlexPrice widget
+/// - [priceLabel]: Text semantic label for [price]
+/// - [oldPriceLabel]: Text semantic label for [oldPrice]
+/// - [enableLineWrap]: Default false: single line display -TextOverflow.ellipsis, True: Enables Line wrap on overlfow
+///
+///
+/// Example usage:
+/// ```dart
+/// FlexPriceDiscount(
+///   price: 99.99,
+///   oldPrice: 200.00,
+///   priceStyle: TextStyle(color: Colors.blue), // Example optional style override
+///   priceFormatter: (price) => NumberFormat.currency(locale: 'en_US').format(price),
+///   priceLabel: "Sale Price {price}" // {price} token position indicates where the post formatter price should be placed in label
+/// )
+/// ```
+///
+class FlexPriceDiscount extends StatelessWidget {
+  const FlexPriceDiscount({
+    super.key,
+    required this.price,
+    this.oldPrice,
+    this.priceFormatter,
+    this.enableLineWrap = false,
+    this.priceLabel,
+    this.priceStyle,
+    this.oldPriceLabel,
+    this.oldPriceStyle,
+  });
+  final double price;
+  final double? oldPrice;
+  final String Function(double)? priceFormatter;
+  final bool enableLineWrap;
+  final String? priceLabel;
+  final String? oldPriceLabel;
+  final TextStyle? priceStyle;
+  final TextStyle? oldPriceStyle;
+
+  @override
+  Widget build(BuildContext context) {
+    final bool displayDiscount =
+        oldPrice != null && oldPrice! > 0 ? true : false;
+
+    final List<Widget> priceWidgets = [
+      if (displayDiscount)
+        FlexPrice(
+          price: oldPrice!,
+          priceFormatter: priceFormatter,
+          priceVariant: PriceVariant.strikethrough,
+          textStyle: const TextStyle(
+            overflow: TextOverflow.ellipsis,
+          ).merge(oldPriceStyle),
+          priceLabel: oldPriceLabel,
+        ),
+      // If single line, spacer between prices elements, otherwise Wrap spacer provided
+      if (!enableLineWrap) const SizedBox(width: 4),
+      FlexPrice(
```

Comment:
Can move these all out of `build()` method too, using helper methods. For example,

```dart
Widget _buildSalePrice() {
    return FlexPrice(
      price: price,
      priceFormatter: priceFormatter,
      textStyle: TextStyle(
        color: context.colors.success,
        overflow: TextOverflow.ellipsis,
      ).merge(priceStyle),
      priceLabel: priceLabel,
    );
  }
```